app/models/pool.rb

```diff
@@ -23,15 +23,28 @@ class Pool < ActiveRecord::Base
 
   DEFAULT_POOL = 'Unassigned'
 
+  def remove_pool
+    return if name == Pool::DEFAULT_POOL
+    reassign_fighters Pool::DEFAULT_POOL
+    pool_number = get_pool_number
+    tournament = Tournament.find(tournament_id)
+    tournament.pools.each do |pool|
```

Comment:
We should make sure this doesn't run too many queries. We may need to use `include` or `join` or something. One way to look at the queries is to start up a server and perform one of these actions that runs the code.